### PR TITLE
This pull request is for tracking the SFX module progress (WIP)

### PR DESCRIPTION
**Pull request by AmirArdroudi**

### CODE REVIEW

File: src/sfx/sfx.cpp [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#include "sfx.hpp"
+#include "soloud.h"
+#include "en/ComponentManager.hpp"
+#include "en/World.hpp"
+
+namespace ari::en
```
Comment (kochol) [inline src/sfx/sfx.cpp:6]: change the namespace to ari::sfx from ari::en::sfx
and write the name space like this
`namespace ari::sfx`

Comment (AmirArdroudi) [reply src/sfx/sfx.cpp:6]: ok sure.

File: src/sfx/sfx.hpp [hunk 2]
```diff
@@ -2,17 +2,14 @@
 
 #include "en/Node.hpp"
 
-namespace ari::en
+namespace ari::sfx
 {
-	namespace sfx
+	class Sfx : public Node
```
Comment (kochol) [inline src/sfx/sfx.hpp:7]: What is this component for?

Comment (AmirArdroudi) [reply src/sfx/sfx.hpp:7]: Other component such as AudioSource and AudioListener would be derived from Sfx and can be behave as a node.

File: src/sfx/AudioSource.hpp [hunk 3]
```diff
@@ -2,13 +2,12 @@
 
 #include "sfx.hpp"
 #include "en/World.hpp"
-
+#include ""
```
Comment (kochol) [inline src/sfx/AudioSource.hpp:5]: Empty include

File: src/sfx/AudioSource.cpp [hunk 4]
```diff
@@ -1,10 +1,19 @@
 #pragma once
 
+#include "sfx.hpp"
 #include"AudioSource.hpp"
-#include "en/ComponentManager.hpp"
 
 namespace ari::sfx
 {
+	SoLoud::Wav wave;
```
Comment (kochol) [inline src/sfx/AudioSource.cpp:8]: ریدی بدم ریدی

File: src/sfx/AudioSource.cpp [hunk 5]
```diff
@@ -1,10 +1,19 @@
...
+		
+
+	AudioSourceHandle LoadAudioSource(char* _path)
```
Comment (kochol) [inline src/sfx/AudioSource.cpp:11]: Add all global functions to sfx.hpp and sfx.cpp

File: src/sfx/sfx.cpp [hunk 6]
```diff
@@ -1,13 +1,15 @@
...
 #include "soloud.h"
-#include "en/ComponentManager.hpp"
-#include "en/World.hpp"
+
+#include "core/log.h"
 
 namespace ari::sfx
 {
-	SoLoud::Soloud* soloud;
+	SoLoud::Soloud* soloud =
```
Comment (kochol) [inline src/sfx/sfx.cpp:8]: New this in InitSfx function

File: src/sfx/AudioSource.cpp (hunk 4 above)
Comment (AmirArdroudi) [reply src/sfx/AudioSource.cpp:8]: Could you be more specific about that? :D

File: src/sfx/sfx.cpp (hunk 6 above)
Comment (AmirArdroudi) [reply src/sfx/sfx.cpp:8]: got it

File: src/sfx/AudioSource.cpp (hunk 5 above)
Comment (AmirArdroudi) [reply src/sfx/AudioSource.cpp:11]: So, Should I remove AudioSource.cpp/.hpp ?

Comment (kochol) [reply src/sfx/AudioSource.cpp:11]: yes

File: src/sfx/sfx.cpp [hunk 7]
```diff
@@ -30,10 +31,11 @@ namespace ari::sfx
 	}
 
 	//------------------------------------------------------------------------------
+	// TODO: Use memory pool 
 	AudioSourceHandle LoadAudioSource(core::String _path)
 	{
 
-		SoLoud::Wav* wave;
+		SoLoud::Wav* wave = new SoLoud::Wav;
```
Comment (kochol) [inline src/sfx/sfx.cpp:38]: use core::Memory::New instead of new

**Pull request closed without merging**